miio/integrations/vacuum/roborock/vacuumcontainers.py

```diff
@@ -94,6 +97,46 @@ def pretty_area(x: float) -> float:
 }
 
 
+class MultiMapList(DeviceStatus):
+    """Contains a information about the maps/floors of the vacuum."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        # {'max_multi_map': 4, 'max_bak_map': 1, 'multi_map_count': 3, 'map_info': [
```

Comment:
Considering we are slowly converting to pydantic (e.g., for genericmiot), I'm wondering should we also do the same here to simplify the parsing and validation?

Reply:
I don't know what pydantic is...
Maybe we can do that in a seperate PR?

Reply:
Okay, so pydantic would allow defining the structure using class annotations like this:
```
class MapInfo(BaseModel):
    flags: int = Field(alias="mapFlag")
    added: datetime = Field(alias="add_time")
    name: Optional[str]  # backup maps contain no name
    backup_maps: Optional[List["MapInfo"]]  # backup maps contain now backups

class MultiMapList(BaseModel):
    maps: List[MapInfo] = Field(alias="map_info")
```
and it would perform the parsing automatically.

The problem I see though is that the DeviceStatus bases itself on the defined properties, so this would need further adjustments... So let's keep it as it is for now.